pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -531,4 +619,20 @@ public FailureDetector.ServerState retryUnhealthyServer(String instanceId) {
 
     return _queryDispatcher.checkConnectivityToInstance(serverInstance);
   }
+
+  public static boolean isYellowError(QueryException e) {
+    switch (e.getErrorCode()) {
+      case QUERY_SCHEDULING_TIMEOUT:
+      case EXECUTION_TIMEOUT:
+      case INTERNAL:
+      case UNKNOWN:
```

Comment:
Why are these "yellow" errors and not "red"?

Reply:
Red errors are errors we didn't catch and somehow bubbled up to the `handleRequest`. By definition, QueryExceptions are never red exceptions. Even if they are internal or unknown, we already caught them and explicitly set a user error message for them.

Colors are not about how problematic the error is. Instead, they are focused on how we want to log the error and how/what send to used.